roles/example-cnf-app/tasks/trex/app.yaml

```diff
@@ -1,105 +1,116 @@
 ---
-- name: set local fact for trex pod networks
+- name: Set local fact for trex pod networks
   set_fact:
     networks_trex: []
     packet_gen_net: "{{ packet_generator_networks if enable_lb|bool else cnf_app_networks }}"
     trex_app_run_passed: false
+
 # TODO(skramaja): This logic to be improved for multiple
 # networks, lets fail it if the length is > 1
 - fail:
     msg: "Need to rewrite the mac merging logic"
   when: "packet_gen_net|length != 1"
-- name: create network list for trex with hardcoded macs
+
+- name: Create network list for trex with hardcoded macs
```

Comment:
agree, it's pushed already :)

```
- name: Wait for TRex app CR to be created
  k8s_info:
    api_version: "examplecnf.openshift.io/v1"
    name: "{{ trex_app_cr_name }}"
    namespace: "{{ cnf_namespace }}"
    kind: "TRexApp"
    wait: true
    wait_condition:
      type: Available
      status: "True"
    wait_timeout: 10
  register: trex_app_cr_installation
  retries: 30
  delay: 10
  until: trex_app_cr_installation is not failed
```

Reply:
and sorry, I had to re-run the job for that, let's see now

Reply:
That should work as well, but I was thinking in doing the `wait` in the k8s creation, that way there's no need to check on it later, but it will not move to the next task until it meets the `wait_condition` required, would that work?

Reply:
It seems like the version of k8s_info we're using is quite old

```
fatal: [jumphost]: FAILED! => {"attempts": 30, "changed": false, "msg": "Unsupported parameters for (k8s_info) module: wait, wait_condition, wait_timeout Supported parameters include: api_key, api_version, ca_cert, client_cert, client_key, context, field_selectors, host, kind, kubeconfig, label_selectors, name, namespace, password, proxy, username, validate_certs"}
```

Probably we have to move to the same version of community.kubernetes.k8s as in d-o-a [there wait etc is supported](https://github.com/redhat-cip/dci-openshift-agent/blame/d7085297e27e24f9b36caeed0b3512e6e369f8cf/roles/deploy-cr/tasks/main.yml#L44-L48).

Reply:
The above makes reference to `k8s_info`, I'm saying we should add it in the `k8s` module, at creation time, that should be supported even if we're using stock k8s:

```
$ ansible-doc k8s 2>&1 | grep -- "- wait"
- wait
- wait_condition
- wait_sleep
- wait_timeout
```

Reply:
Hi @tonyskapunk, no, waiting in k8s makes no sense unless we introduce complex control variables.

```
# k8s part
# create CR, the output will show "created" immediately and fire the installation
$ oc apply -f my-nice-deployment.yml
trexapp.examplecnf.openshift.io/trex-app1 created

# k8s_info part
# wait for CR to be created
$ oc get trexapp.examplecnf.openshift.io -o yaml -n example-cnf trex-app1
apiVersion: examplecnf.openshift.io/v1
kind: TRexApp
metadata:
  annotations:
    ansible.sdk.operatorframework.io/reconcile-period: 1m
    ansible.sdk.operatorframework.io/verbosity: "4"
  creationTimestamp: "2022-10-13T20:45:37Z"
  generation: 2
  name: trex-app1
  namespace: example-cnf
  resourceVersion: "187964"
  uid: ea0bff7d-9cad-43f1-b98f-897c1277ee3d
spec:
  duration: 120
  enableLb: true
  imagePullPolicy: Always
  lbMacs:
  - 40:04:0f:f1:89:01
  - 40:04:0f:f1:89:02
  org: rh-nfv-int
  packetRate: 10kpps
  packetSize: 64
  registry: quay.io
  trexApp: false
  version: v0.2.3
status:
  conditions:
  - ansibleResult:
      changed: 0
      completion: 2022-10-13T20:58:51.933846
      failures: 0
      ok: 7
      skipped: 2
    lastTransitionTime: "2022-10-13T20:45:37Z"
    message: Awaiting next reconciliation
    reason: Successful
    status: "True"
    type: Running
```

And unfortunately, for the current version of k8s_info wait is not supported, [here is the failing test](https://www.distributed-ci.io/jobs/1ac57ada-d34e-461b-ac12-d3ac52b101b4/jobStates?sort=date&task=eb83da2e-b8e7-4a70-bf4d-16d3874565b7): 
```
fatal: [jumphost]: FAILED! => {"attempts": 30, "changed": false, "msg": "Unsupported parameters for (k8s_info) module: wait, wait_condition, wait_timeout Supported parameters include: api_key, api_version, ca_cert, client_cert, client_key, context, field_selectors, host, kind, kubeconfig, label_selectors, name, namespace, password, proxy, username, validate_certs"}
```

I'm going to decouple k8s dependencies and a fix for TRex CR (wait for CR and introduce pods security).
k8s dependencies could definitely be addressed as the next step.

Reply:
> I'm going to decouple k8s dependencies and a fix for TRex CR (wait for CR and introduce pods security). k8s dependencies could definitely be addressed as the next step.

I suggest to handle pod security in a separated change, that is only affecting 4.12+ and the wait CR is something that any version of OCP would benefit, my 2¢